Python_Backend/AnswerGenerator.py

```diff
@@ -28,62 +37,190 @@ def clear_queue(self):
 
 
     @abc.abstractmethod
-    def add_state_to_queue(self):
+    def add_frame_to_queue(self):
         """Removes all elements from the queue"""
         raise NotImplementedError
 
 
-    @abc.abstractmethod
-    def determine_answer(self):
-        """looks at the current queue of states and passes along a yes/no or raises an error"""
-        raise NotImplementedError
 
+class IrisAnswerGenerator(AnswerGeneratorInterface):
+
+    # Tuning Variables
+    NUM_OF_UP_DOWN_PATTERNS_FOR_YES   = 3
+    NUM_OF_LEFT_RIGHT_PATTERNS_FOR_NO = 3
 
+    FRAME_QUEUE_SIZE = 30
+    STATE_QUEUE_SIZE = max(NUM_OF_LEFT_RIGHT_PATTERNS_FOR_NO, NUM_OF_UP_DOWN_PATTERNS_FOR_YES) * 2
 
+    # Required number of frames in __past_frames queue for state to be added to __past_states queue
+    FRAMES_FOR_STATE_LIST = [25,  # eyes_up
+                             25,  # eyes_down

```

Comment:
are these minimum values? or exact values? are we able to have play here? As in it could be + or - 3 frames so a range from 22 to 28?

Reply:
These are minimum values not exact, so since the FRAME_QUEUE_SIZE = 30, and the minimum frames are 25, 25-30/30 frames would all work.